tx-indexer/src/collector.rs

```diff
@@ -292,11 +293,57 @@ async fn save_transaction_details(
         }
     };
     let transaction_hash = transaction_details.transaction.hash.to_string();
+    let signer_id = transaction_details.transaction.signer_id.to_string();
+    let tx_bytes = &borsh::to_vec(&transaction_details).expect("Failed to serialize transaction");
+
     match db_manager
-        .add_transaction(transaction_details, tx_details.block_height)
+        .add_transaction(
+            &transaction_hash,
+            tx_bytes.clone(),
+            tx_details.block_height,
+            &signer_id,
+        )
         .await
     {
         Ok(_) => {
+            let mut attempts = 0;
+            let mut tx_bytes = tx_bytes.clone();
+            while let Err(err) = db_manager
+                .validate_saved_transaction_deserializable(&transaction_hash, &tx_bytes)
+                .await
+            {
+                tracing::warn!(
+                    target: crate::INDEXER,
+                    "Failed to validate transaction {} \n{:#?}",
+                    transaction_hash,
+                    err
+                );
+                if attempts >= 3 {
+                    return false;
```

Comment:
Please add a warning here with the transaction hash so we can see in logs that we attempted 3 times and failed